lib/sensu/server/process.rb

```diff
@@ -552,6 +586,7 @@ def determine_stale_clients
             @redis.get("client:#{client_name}") do |client_json|
               unless client_json.nil?
                 client = MultiJson.load(client_json)
+                next if client[:keepalives] == false
```

Comment:
Correct, keepalive events will never be created for these clients, therefor, deregistration handlers will never be executed. Manually deleting a client will resolve current events and cleanup history etc \o/